bundle/src/main/java/com/adobe/acs/commons/mcp/util/Variant.java

```diff
@@ -0,0 +1,271 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2018 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.mcp.util;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Optional;
+import java.text.ParseException;
+import org.apache.commons.lang.StringUtils;
+import org.apache.poi.ss.usermodel.Cell;
+
+/**
+ * Used to represent values that might be provided as one type but used as another. Avoids glue code and switch
+ * statements in other parts of the code expecially dealing with data from spreadsheets.
+ */
+public class Variant {
```

Comment:
Should be final I think. And marked as @ProviderType (that should cause the package export to only be minor).

Reply:
Same question about whether or not this needs to be API

Reply:
The packaging implies it would/should be used in context of MPC tools — you’re going to create MPC tooling outside of commons that use this? 

Perhaps this should be moved to a more generic package space if you plan on generally using it as a dep outside commons?

Reply:
I'm all for it -- Spreadsheet parsing and data handling is a more generic concern than MCP.   Any suggestions?

Reply:
commons.data?